cdap-watchdog/src/main/java/io/cdap/cdap/metrics/process/MessagingMetricsProcessorManagerService.java

```diff
@@ -139,6 +144,28 @@ protected void startUp() throws Exception {
     }
   }
 
+  @VisibleForTesting
+  void initializeMetricWriter(MetricsWriter writer,
+                                      DefaultMetricsWriterContext metricsWriterContext) throws Exception {
+    File initStateFile = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR), writer.getID());
```

Comment:
would be better to add a directory here, so it goes in`local_data_dir/metricswriters/writer.getId()`. That way in case anything else needs to write files, there isn't a chance of overlap

Reply:
Done